crates/stages/src/trie/mod.rs

```diff
@@ -0,0 +1,581 @@
+use crate::Transaction;
```

Comment:
The DB-backed cita_trie impl + loader look good, nice work on the tests. I'm worried about performance here slightly.

Reply:
At the start we were using parity's `trie-db`, but we [changed it](https://github.com/paradigmxyz/reth/pull/994/commits/40b62ffd6dc8f2d10248bf06e49816662cbdb05b) because of root hash mismatching (see https://github.com/paritytech/trie/issues/182), along with non-existent error handling in their [database trait](https://github.com/paritytech/trie/blob/master/hash-db/src/lib.rs#L126-L147). The idea is to change the loader to use [ours](https://github.com/lambdaclass/merkle_patricia_tree) when it's finished (pending benchmarks, and optionally adding some way to more easily change implementations).